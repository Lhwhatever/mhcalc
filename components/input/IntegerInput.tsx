import { TextField } from '@material-ui/core'
import React from 'react'
import { coerceToRange } from '../../utils/number'
import { asInteger } from '../../utils/str'
import { InputChangeEvent } from './types'

export interface NumericInputProps {
    onChange: (newValue: number | undefined) => void
    value?: number
    min?: number
    max?: number
}

const IntegerInput = (props: NumericInputProps): JSX.Element => {
    const { onChange, min, max, value, ...other } = props

    const handleChange = (event: InputChangeEvent) => {
        if (event.target.value === '') onChange(undefined)
        if (/^-?\d+$/.test(event.target.value)) onChange(asInteger(event.target.value))
    }

    const handleUnfocus = () => {
        if (value === undefined || value < min || value > max) onChange(coerceToRange(value ?? 0, min, max))
    }

    return (
        <TextField type="number" {...other} onChange={handleChange} onBlur={handleUnfocus} inputProps={{ min, max }} />
    )
}

export default IntegerInput
