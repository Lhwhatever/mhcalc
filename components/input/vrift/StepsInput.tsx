import React from 'react'
import IntegerInput, { IntegerInputProps } from '../IntegerInput'

export type StepsInputProps = Omit<IntegerInputProps, 'min' | 'max'>

const StepsInput = (props: StepsInputProps): JSX.Element => {
    return <IntegerInput label="Steps" variant="outlined" min={0} {...props} />
}

export default StepsInput
