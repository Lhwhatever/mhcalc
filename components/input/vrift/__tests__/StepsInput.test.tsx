import { TextField } from '@material-ui/core'
import { createShallow } from '@material-ui/core/test-utils'
import React from 'react'
import StepsInput from '../StepsInput'
import IntegerInput from '../../IntegerInput'

describe('StepsInput test', () => {
    let shallow
    let wrapper

    beforeAll(() => {
        shallow = createShallow()
    })

    beforeEach(() => {
        wrapper = shallow(<StepsInput onChange={jest.fn()} />)
    })

    it('should use IntegerInput[label="Steps"]', () => {
        expect(wrapper.find(IntegerInput)).toHaveLength(1)
        expect(wrapper.find(IntegerInput).prop('label')).toMatch(/steps/i)
    })

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
