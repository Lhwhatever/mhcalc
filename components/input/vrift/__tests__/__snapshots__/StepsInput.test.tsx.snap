// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepsInput test should match snapshot 1`] = `
<IntegerInput
  label="Steps"
  min={0}
  onChange={[MockFunction]}
  variant="outlined"
/>
`;
